integration-auth/src/main/java/org/symphonyoss/integration/authentication/AuthenticationContext.java

```diff
@@ -50,15 +66,47 @@
 
   public AuthenticationContext(String userId, KeyStore keyStore, String keyStorePass) {
     this.userId = userId;
+    this.client = buildClient(keyStore, keyStorePass);;
+  }
 
+  private Client buildClient(KeyStore keyStore, String keyStorePass) {
```

Comment:
Does this have anything to do with the memory leak or is it just tuning the client for more throughput?

Have you tested before/after?